pkg/panels/blackbox/blackbox.go

```diff
@@ -0,0 +1,777 @@
+package blackbox
+
+import (
+	"github.com/perses/community-dashboards/pkg/dashboards"
+	"github.com/perses/community-dashboards/pkg/promql"
+	commonSdk "github.com/perses/perses/go-sdk/common"
+	"github.com/perses/perses/go-sdk/panel"
+	panelgroup "github.com/perses/perses/go-sdk/panel-group"
+	stat "github.com/perses/perses/go-sdk/panel/stat"
+	timeSeriesPanel "github.com/perses/perses/go-sdk/panel/time-series"
+	"github.com/perses/perses/go-sdk/prometheus/query"
+)
+
+// ProbeStatusMapfunc creates a panel option for displaying Blackbox Probe Success for all instances
+//
+// The panel uses the following Prometheus metrics:
+// - probe_success: indicates if the probe succeeded
+//
+// The panel shows:
+// - Probe success, either 1 if up, or 0 if down
+//
+// Parameters:
+//   - datasourceName: The name of the Prometheus data source.
+//   - labelMatchers: A variadic parameter for Prometheus label matchers to filter the query.
+//
+// Returns:
+//   - panelgroup.Option: A panel option that can be added to a panel group.
+
+func ProbeStatusMapfunc(datasourceName string, labelMatchers ...promql.LabelMatcher) panelgroup.Option {
+	return panelgroup.AddPanel("Status Map",
+		panel.Description("Shows Probe success, either 1 if up, or 0 if down"),
+		stat.Chart(
+			stat.Calculation(commonSdk.LastCalculation),
+			stat.Format(commonSdk.Format{
+				Unit: string(commonSdk.DecimalUnit),
+			}),
+			stat.WithSparkline(stat.Sparkline{
+				Width: 1,
+			}),
+			stat.Thresholds(commonSdk.Thresholds{
+				Mode:         commonSdk.AbsoluteMode,
+				DefaultColor: "green",
+				Steps: []commonSdk.StepOption{
+					{
+						Color: "red",
+						Value: 0,
+					},
+					{
+						Color: "green",
+						Value: 1,
+					},
+				},
+			}),
+		),
+		panel.AddQuery(
+			query.PromQL(
+				promql.SetLabelMatchers(
+					"max by (instance) (probe_success{job=~'$job'})",
+					labelMatchers,
+				),
+				dashboards.AddQueryDataSource(datasourceName),
+				query.SeriesNameFormat("{{instance}}"),
+			),
+		),
+	)
+}
+
+// ProbeSuccessCount creates a panel option for displaying a counter of all Probe Success
+//
+// The panel uses the following Prometheus metrics:
+// - probe_success: indicates if the probe succeeded
+//
+// The panel shows:
+// - Counter of all existing probe_success, so the number of endpoints being probed
+//
+// Parameters:
+//   - datasourceName: The name of the Prometheus data source.
+//   - labelMatchers: A variadic parameter for Prometheus label matchers to filter the query.
+//
+// Returns:
+//   - panelgroup.Option: A panel option that can be added to a panel group.
+func ProbeSuccessCount(datasourceName string, labelMatchers ...promql.LabelMatcher) panelgroup.Option {
+	return panelgroup.AddPanel("Probes",
+		panel.Description("Counts Probes Success"),
+		stat.Chart(
+			stat.Calculation(commonSdk.LastCalculation),
+			stat.Format(commonSdk.Format{
+				Unit: string(commonSdk.DecimalUnit),
```

Comment:
Same here, we always gonna have integers here, so I think we don't really need a precision